infraci.jenkins.io-kubernetes-sponsored-agents.tf

```diff
@@ -59,3 +59,94 @@ resource "azurerm_kubernetes_cluster" "infracijenkinsio_agents_1" {
 
   tags = local.default_tags
 }
+
+
+# Node pool to host "jenkins-infra" applications required on this cluster such as datadog's cluster-agent, e.g. "Not agent, neither AKS System tools"
+resource "azurerm_kubernetes_cluster_node_pool" "linux_arm64_n2_applications_sponsorship" {
+  provider              = azurerm.jenkins-sponsorship
+  name                  = "la64n2app"
+  vm_size               = "Standard_D4pds_v5"
+  os_disk_type          = "Ephemeral"
+  os_disk_size_gb       = 150 # Ref. Cache storage size at https://learn.microsoft.com/en-us/azure/virtual-machines/dv3-dsv3-series#dsv3-series (depends on the instance size)
+  orchestrator_version  = local.kubernetes_versions["infracijenkinsio_agents_1"]
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.infracijenkinsio_agents_1.id
+  enable_auto_scaling   = true
+  min_count             = 1
+  max_count             = 3 # 2 nodes always up for HA, a 3rd one is allowed for surge upgrades
+  zones                 = local.infracijenkinsio_agents_1_compute_zones
+  vnet_subnet_id        = data.azurerm_subnet.infraci_jenkins_io_kubernetes_agent_sponsorship.id
+
+  node_labels = {
+    "jenkins" = "infra.ci.jenkins.io"
+    "role"    = "applications"
+  }
+  node_taints = [
+    "infra.ci.jenkins.io/applications=true:NoSchedule",
+  ]
+
+  lifecycle {
+    ignore_changes = [node_count]
+  }
+
+  tags = local.default_tags
+}
```

Comment:
```suggestion
```

pair review: the only expected "application" is the Datadog's cluster-agent (2 pods). We should reuse the system pool to run it as it's not an heavy consumer: confirmed by checking the load on the ci.jenkins.io-agents-1 cluster with the same kind of node pool.

@smerle33 to comment the issue to cherry-pick this choice and arguments